tests/test_args.py

```diff
@@ -73,266 +77,282 @@ def log_capture(caplog, args):
     return LogCaptureObject(caplog.record_tuples, e)
 
 
-def test_version(capsys):
+def test_version(capsys: pytest.CaptureFixture[str]) -> None:
```

Comment:
Type annotations on fixtures and test functions are less helpful because it cannot be typechecked that the given type matches the fixture's actual type. In the projects I work on, I therefore exclude test files from the Mypy `disallow_untyped_defs` check:

```toml
[[tool.mypy.overrides]]
module = "tests"
disallow_untyped_defs = false

```

see https://mypy.readthedocs.io/en/stable/config_file.html#using-a-pyproject-toml-file

Reply:
I missed this. I'll dress this in a separate PR.

Reply:
This doesn't work because this only works on module and not on folder level, see https://github.com/python/mypy/issues/17977. Shall we exclude the whole test directory?

Reply:
Should work if there's a `tests/__init__.py` file, which turns the `tests` directory into a module (as far as Mypy is concerned). Gcovr has such a file, surprised that it doesn't work. Hmm. I'll try to look into it when I find the time.

Reply:
My fault. We also need to set `disallow_incomplete_defs = false`.